libnetdata/socket/socket.c

```diff
@@ -1044,21 +1112,13 @@ int accept_socket(int fd, int flags, char *client_ip, size_t ipsize, char *clien
                 debug(D_LISTENER, "New UNKNOWN web client from %s port %s on socket %d.", client_ip, client_port, fd);
                 break;
         }
-
-        if(access_list) {
-            if(!strcmp(client_ip, "127.0.0.1") || !strcmp(client_ip, "::1")) {
-                strncpy(client_ip, "localhost", ipsize);
-                client_ip[ipsize - 1] = '\0';
-            }
-
-            if(unlikely(!simple_pattern_matches(access_list, client_ip))) {
-                errno = 0;
-                debug(D_LISTENER, "Permission denied for client '%s', port '%s'", client_ip, client_port);
-                error("DENIED ACCESS to client '%s'", client_ip);
-                close(nfd);
-                nfd = -1;
-                errno = EPERM;
-            }
+        if(!connection_allowed(&sadr, addrlen, client_ip, ipsize, access_list)) {
+            errno = 0;
+            debug(D_LISTENER, "Permission denied for client '%s', port '%s'", client_ip, client_port);
+            error("DENIED ACCESS to client '%s'", client_ip);
```

Comment:
Since you're already logging an error, you can lose the debug log and just add the port to the error message, if you feel it's needed.

Reply:
Fixed.